### ISSUE TITLE
Add `supportedQueryInterfaces` support to PanelPlugin interface

### ISSUE THREAD

**Issue by sjcobb** [perses/perses#1303]
See 'Decoupling data querying from panel plugin' discussion #956, this feature is follow up now that we have `DataQueriesProvider`. 

Currently, we only have TimeSeriesQuery plugins that are used in our core panels (TimeSeriesChart, GaugeChart, StatChart). We also want the ability to add more query plugins such as TraceQuery and LogQuery. To support these we need a way for a panel plugin to specify which interfaces it supports (see PR #922 for more info in [comment here](https://github.com/perses/perses/pull/922#discussion_r1080546854))

```ts
export interface PanelPlugin<Spec = UnknownSpec> extends Plugin<Spec> {
  PanelComponent: React.ComponentType<PanelProps<Spec>>;
  supportedQueryInterfaces?: Array<'TimeSeriesQuery' | 'LogQuery' | 'TraceQuery'>; // NEW
  panelOptionsEditorComponents?: Array<PanelOptionsEditorComponent<Spec>>;
  hideQueryEditor?: boolean; // can be removed 
}
```

In the example above, a future Table panel plugin would support all query interfaces, while TimeSeriesChart would not. hideQueryEditor can be removed and inferred when supportedQueryInterfaces is empty Array

Also, relates to #1151 for adding instant query support since a new query plugin is the ideal way to implement that functionality.

**Comment by eunicorn** [comment]
After #1318 is merged, we can introduce “supportedQueryInterface” in panel plugins. The idea here is to give the framework the ability to render the appropriate query editor depending on which panel plugin is selected, enhancing dashboard user experience. For example, if user selects a time series chart, our panel editor will know to render a time series query editor. 

Additionally, the panel plugin can determine the active query interface to dynamically adjust its options. To do this, we can have `PanelEditor` becomes the children of `DataQueriesProvider`. This way, the options editor components can access `useDataQueries` or other relevant hooks to identify the data type returned by the query.

**Comment by celian-garcia** [comment]
Hello! if you don´t mind I'm gonna take this one this week ;) I will ask you directly if I face any issue. Assigning to myself.

**Comment by celian-garcia** [comment]
Regarding the PR upper 👆, I will close it and split it into three parts in order to be more readable and more feasible for me to implement 
- :one: the `PluginEditor` should support several query types`` (this is almost done in that PR) 
![image](https://github.com/perses/perses/assets/10258608/58e02f9d-d1d3-45ff-80e5-84bfd043cdaf)
- :two: build a `QueryEditor` like `TimeSeriesQueryEditor` or `TraceQueryEditor` but that takes the QueryType as parameter (let's create a separate one and deprecate the two others) 
![image](https://github.com/perses/perses/assets/10258608/b26f7981-d99b-43e1-82f5-e994a2f18002)
- :three:  Introduce the `supportedQueryInterfaces` (or `supportedQueryTypes` , or `queryTypes`) field and use it in the `PanelSpecEditor` to trigger usage of the new `QueryEditor`

Then it would be up to the plugin developer to use the generic `QueryEditor` or not through this new field.
